### PR TITLE
feat(vrack-services): use manager-components

### PR DESCRIPTION
**Pull request by chipp972**
ref: MANAGER-14229

| Question         | Answer
| ---------------- | ---
| Branch?          | `develop` 
| Bug fix?         | no
| New feature?     | yes
| Breaking change? | no
| Tickets          | MANAGER-14229 
| License          | BSD 3-Clause

- [x] Try to keep pull requests small so they can be easily reviewed.
- [x] Commits are signed-off
- [ ] ~~Only FR translations have been updated~~
- [x] Branch is up-to-date with target branch
- [x] Lint has passed locally
- [x] Standalone app was ran and tested locally
- [x] Ticket reference is mentioned in linked commits (internal only)
- [ ] ~~Breaking change is mentioned in relevant commits~~

### CODE REVIEW

File: packages/manager-components/src/utils/click-utils.ts [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+ * Assign the function passed to the onClick event handler
+ * as welll as the onKeyDown event handler when using the keys Space or Enter
+ */
+export const handleClick = (
+  fn: React.KeyboardEventHandler & React.MouseEventHandler,
+) => ({
+  onClick: fn,
+  onKeyDown: (event: React.KeyboardEvent) => {
+    if ([' ', 'Enter'].includes(event.key)) {
+      fn(event as React.KeyboardEvent & React.MouseEvent);
```
Comment (frenautvh) [inline packages/manager-components/src/utils/click-utils.ts:13]: not really fond of casting keyboard event to mouseevent, if you are only interested in "click" event you could define a custom handler type for your "fn" function, maybe something like :

```ts
fn: (ev: Event) => void
```

using a more generic "Event" type make more sense

Comment (chipp972) [reply packages/manager-components/src/utils/click-utils.ts]: Thank for your feedback

**Review by aboungnaseng-ovhcloud — APPROVED** [review]

File: packages/manager/apps/vrack-services/src/pages/subnets/Listing/SubnetDataGrid.tsx [hunk 2]
```diff
@@ -0,0 +1,194 @@
+/* eslint-disable import/prefer-default-export */
```
[the remark this answers is not in the document]
Comment (chipp972) [reply packages/manager/apps/vrack-services/src/pages/subnets/Listing/SubnetDataGrid.tsx]: done

File: packages/manager/apps/vrack-services/src/pages/endpoints/Listing/EndpointDataGrid.tsx [hunk 3]
```diff
@@ -0,0 +1,170 @@
+/* eslint-disable import/prefer-default-export */
```
[the remark this answers is not in the document]
Comment (chipp972) [reply packages/manager/apps/vrack-services/src/pages/endpoints/Listing/EndpointDataGrid.tsx]: done

File: packages/manager-components/src/components/templates/delete-modal/delete-modal.component.tsx [hunk 4]
```diff
@@ -19,10 +20,10 @@ import {
...
   ODS_TEXT_SIZE,
 } from '@ovhcloud/ods-components';
-import { ApiError } from '@ovh-ux/manager-core-api';
-import { handleClick } from '@/utils/ods-utils';
-import { FormField } from '../FormField';
-import { terminateValue } from './DeleteModal.constants';
+import { handleClick } from '../../../utils/click-utils';
+import './translations/translations';
+
+const terminateValue = 'TERMINATE';
```
Comment (qpavy) [inline packages/manager-components/src/components/templates/delete-modal/delete-modal.component.tsx:26]: Maybe put this in the props to change it, if needed to change it from the component

File: packages/manager/apps/vrack-services/src/App.tsx [hunk 5]
```diff
@@ -28,7 +30,18 @@ export const App: React.FC = () => {
 
   return (
     <QueryClientProvider client={queryClient}>
-      <RouterProvider router={router} />
+      <UpdateDisplayNameContext.Provider
```
Comment (qpavy) [inline packages/manager/apps/vrack-services/src/App.tsx:33]: Shouldn't we have a context for all type of messages ? Not only for display name messages ?

Comment (chipp972) [reply packages/manager/apps/vrack-services/src/App.tsx:33]: https://github.com/ovh/manager/pull/11781/files#diff-ca8098eaca001628500d33defaea5d2f4a33e0a2a79767bf211f264cd593a6f9

Will be done in the next PR

File: packages/manager-components/src/components/templates/delete-modal/delete-modal.component.tsx (hunk 4 above)
Comment (chipp972) [reply packages/manager-components/src/components/templates/delete-modal/delete-modal.component.tsx]: Thanks done

**Review by qpavy — APPROVED** [review]

**Pull request merged**